cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381/Internal.hs

```diff
@@ -428,6 +459,17 @@ withNewScalar_ = fmap fst . withNewScalar
 withNewScalar' :: (ScalarPtr -> IO a) -> IO Scalar
 withNewScalar' = fmap snd . withNewScalar
 
+withScalarVector :: NonEmpty.NonEmpty Scalar -> (ScalarPtrVector -> IO a) -> IO a
+withScalarVector scalars go = do
+  let numScalars = NonEmpty.length scalars
+  let sizeReference = sizeOf (undefined :: Ptr ())
+  allocaBytes (numScalars * sizeReference) $ \ptr -> do
+    let copyPtrAtIx ix scalar =
+          withScalar scalar $ \(ScalarPtr sPtr) -> do
+          poke (ptr `advancePtr` ix) sPtr
```

Comment:
Wouldn't this leak the address in `sPtr` outside the `withScalar` block?

Reply:
Sharp! The same is true for the vector of pointers that is built in the `withAffineVector` function